components/script/dom/bindings/str.rs

```diff
@@ -303,6 +304,37 @@ impl DOMString {
         parse_week_string(&*self.0).is_ok()
     }
 
+    /// https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#valid-floating-point-number
+    /// The rust default String::parse<f64>() is extremely close, and this just fills in the gaps.
+    pub fn is_valid_number_string(&self) -> bool {
+        let input = &self.0;
+        let parsed_result = input.parse::<f64>();
+        match parsed_result {
+            Ok(val) => {
+                if val.is_infinite() {
+                    return false;
+                }
+                // Edge case. Rust's formatter will interpret "1." as 1,
+                // but we want that to be invalid.
+                if input.ends_with(".") {
+                    return false;
+                }
+                // Another edge case. Rust interprets the literal "NaN" to be NaN
+                if val.is_nan() {
+                    return false;
+                }
+                // Finally, if the value starts with "+", it is invalid
+                if input.starts_with("+") {
+                    return false;
+                }
+                return true;
```

Comment:
I guess no need `return` here?

Reply:
:+1: . I guess I'm still used to C++ with it's explicit return.